framework/utils/mathUtils.py

```diff
@@ -33,6 +33,9 @@
 import MessageHandler # makes sure getMessageHandler is defined
 mh = getMessageHandler()
 
+# dict of builtin types, filled by getBuiltinTypes
+_bTypes = None
```

Comment:
several questions:

Even if you are defining _bTypes here, It seems you are not using it in the ```checkTypeHDF5```, right?
Second, since you have defined the global ```_bTypes```,  it may worth to define the global ```_nTypes``` for numpy types.

Reply:
I use it in the ```getBuiltinTypes``` that is used in ```checkTypeHDF5```

For the numpy one I do not do it because numpy.sctypes is already a dictionary.